src/backend/base/langflow/components/models/HuggingFaceModel.py

```diff
@@ -18,22 +20,81 @@ class HuggingFaceEndpointsComponent(LCModelComponent):
             display_name="Model ID",
             value="openai-community/gpt2",
         ),
+        StrInput(
+            name="inference_endpoint",
+            display_name="Inference Endpoint",
+            value="https://api-inference.huggingface.co/models/",
+            info="Custom inference endpoint URL.",
+        ),
         DropdownInput(
             name="task",
             display_name="Task",
             options=["text2text-generation", "text-generation", "summarization", "translation"],
-            value="text-generation",
+            advanced=True,
+            info="The task to call the model with. Should be a task that returns `generated_text` or `summary_text`.",
         ),
         SecretStrInput(name="huggingfacehub_api_token", display_name="API Token", password=True),
         DictInput(name="model_kwargs", display_name="Model Keyword Arguments", advanced=True),
         IntInput(name="retry_attempts", display_name="Retry Attempts", value=1, advanced=True),
+        IntInput(
+            name="max_new_tokens", display_name="Max New Tokens", value=512, info="Maximum number of generated tokens"
+        ),
+        IntInput(
+            name="top_k",
+            display_name="Top K",
+            advanced=True,
+            info="The number of highest probability vocabulary tokens to keep for top-k-filtering",
+        ),
+        FloatInput(
+            name="top_p",
+            display_name="Top P",
+            value=0.95,
+            advanced=True,
+            info="If set to < 1, only the smallest set of most probable tokens with probabilities that add up to `top_p` or higher are kept for generation",
+        ),
+        FloatInput(
+            name="typical_p",
+            display_name="Typical P",
+            value=0.95,
+            advanced=True,
+            info="Typical Decoding mass. See [Typical Decoding for Natural Language Generation](https://arxiv.org/abs/2202.00666) for more information",
+        ),
+        FloatInput(
+            name="temperature",
+            display_name="Temperature",
+            value=0.8,
+            advanced=True,
+            info="The value used to module the logits distribution",
+        ),
+        FloatInput(
+            name="repetition_penalty",
+            display_name="Repetition Penalty",
+            advanced=True,
+            info="The parameter for repetition penalty. 1.0 means no penalty. See [this paper](https://arxiv.org/pdf/1909.05858.pdf) for more details",
```

Comment:
I'm not sure if we need to be linking papers like this here and above

Reply:
I have removed it.
I initially  took it from the Comments of langchain docs of huggingface,